src/binary.ts

```diff
@@ -311,42 +310,40 @@ export class Binary extends BSONValue {
 export type UUIDExtended = {
   $uuid: string;
 };
+
 const UUID_BYTE_LENGTH = 16;
+const UUID_WITHOUT_DASHES = /^[0-F]{32}$/i;
+const UUID_WITH_DASHES = /^[0-F]{8}-[0-F]{4}-[0-F]{4}-[0-F]{4}-[0-F]{12}$/i;
 
 /**
  * A class representation of the BSON UUID type.
  * @public
  */
 export class UUID extends Binary {
-  static cacheHexString: boolean;
-
-  /** UUID hexString cache @internal */
-  private __id?: string;
-
+  static cacheHexString = false;
   /**
-   * Create an UUID type
+   * Create a UUID type
+   *
+   * When the argument to the constructor is omitted a random v4 UUID will be generated.
    *
    * @param input - Can be a 32 or 36 character hex string (dashes excluded/included) or a 16 byte binary Buffer.
    */
   constructor(input?: string | Uint8Array | UUID) {
     let bytes: Uint8Array;
-    let hexStr;
     if (input == null) {
       bytes = UUID.generate();
     } else if (input instanceof UUID) {
       bytes = ByteUtils.toLocalBufferType(new Uint8Array(input.buffer));
-      hexStr = input.__id;
     } else if (ArrayBuffer.isView(input) && input.byteLength === UUID_BYTE_LENGTH) {
```

Comment:
I brought it up in the ticket but this can still lead to validation issues:

```js
const { Binary, serialize, deserialize } = require('bson');

const bin = Binary.createFromHexString('aaaaaa', 4);
const s = serialize({ bin });
const d = deserialize(s); // throws
```

and of course this can also happen when reading data from a real server. (My suggestion would be to skip this [branch here](https://github.com/mongodb/js-bson/blob/91b84ed72c1f1f5f882d993c2fec9a4aea9740a9/src/parser/deserializer.ts#L407-L408) for invalid UUIDs but I don’t have strong feelings about that.)

If that’s not in scope for this ticket, I’d make sure to create another one.

Reply:
Good catch, I was only thinking about 16 byte sequences we were too strictly validating. I think we should make this change, the spirit of the ticket is to handle values generated outside this library more gracefully.